zrml/prediction-markets/src/lib.rs

```diff
@@ -3442,7 +3447,11 @@ mod pallet {
                 Asset::Ztg => true,
                 #[cfg(feature = "parachain")]
                 Asset::ForeignAsset(fa) => {
-                    if let Some(metadata) = T::AssetRegistry::metadata(&Asset::ForeignAsset(fa)) {
+                    if let Some(metadata) =
+                        T::AssetRegistry::metadata(&CurrencyClass::<MarketIdOf<T>>::ForeignAsset(
+                            fa,
+                        ))
+                    {
```

Comment:
[Old structure](https://github.com/zeitgeistpm/zeitgeist/blob/4628f8c0af5f44eab01add840fee8e7b78aced1e/primitives/src/assets/currencies.rs#L21-L42) for orml_asset_registry used here, because we store the old structure in storage of orml_asset_registry, right?

Reply:
`CurrencyClass` is the new and only structure used in orml-tokens. `Old` variants are only supported due to backward compatibility. 
AssetRegistry` only has `Asset::ForeignAsset(id)` (scale index 5) in it's storage. Anything that is read from the `AssetRegistry` will result in `CurrencyClass<MarketId>::ForeignAsset(id)` (or in short, `Currencies::ForeignAsset(id)` (scale index 5).